### PR TITLE
Make StringComparer.Create throw ArgumentNullException

### PR DESCRIPTION
**Pull request by TylerBrinkley**
Throws proper exception.

@danmosemsft

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/StringComparer.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public static StringComparer Create(CultureInfo culture, CompareOptions options)
         {
             if (culture == null)
             {
-                throw new ArgumentException(nameof(culture));
+                throw new ArgumentNullException(nameof(culture));
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/StringComparer.cs]: A test for this is conspicuously missing from https://github.com/dotnet/corefx/blob/8455fd2d9bd46fc8b97fe86f907d2a8ae748443f/src/System.Runtime/tests/System/StringComparerTests.cs#L13-L16.  Would you mind opening a PR in corefx coverage for this?  Thanks.

Comment (TylerBrinkley) [reply src/System.Private.CoreLib/shared/System/StringComparer.cs]: Created as https://github.com/dotnet/corefx/pull/40902

**Comment by danmosemsft** [comment]
Oh, we forgot it is necessary to update https://github.com/dotnet/coreclr/blob/master/tests/CoreFX/CoreFX.issues.rsp to make the tests pass. That is because it is running the CoreFX tests that do not have your change (and it appears there _is_ a test for it)

```
Assert.Throws() Failure\r\nExpected: typeof(System.ArgumentException)\r\nActual: typeof(System.ArgumentNullException): Value cannot be null. (Parameter 'culture')\r\n---- System.ArgumentNullException : Value cannot be null. (Parameter 'culture')

Stack trace
   at System.StringComparer.Create(CultureInfo culture, CompareOptions options) in /_/src/System.Private.CoreLib/shared/System/StringComparer.cs:line 64
   at System.Tests.StringComparerTests.<>c.<CreateCultureOptions_InvalidArguments_Throws>b__7_0() in /_/src/System.Runtime/tests/System/StringComparerTests.netcoreapp.cs:line 15
----- Inner Stack Trace -----
   at System.StringComparer.Create(CultureInfo culture, CompareOptions options) in /_/src/System.Private.CoreLib/shared/System/StringComparer.cs:line 64
   at System.Tests.StringComparerTests.<>c.<CreateCultureOptions_InvalidArguments_Throws>b__7_0() in /_/src/System.Runtime/tests/System/StringComparerTests.netcoreapp.cs:line 15
```

**Comment by danmosemsft** [comment]
We're planning to merge the corefx/coreclr repos so this coordination problem will go away at that point.

**Pull request merged**